crates/p2p/src/swarm/mod.rs

```diff
@@ -0,0 +1,713 @@
+//! Swarm implementation for P2P.
+
+use std::{
+    collections::HashSet,
+    sync::LazyLock,
+    time::{Duration, Instant},
+};
+
+use behavior::{Behaviour, BehaviourEvent};
+use errors::{P2PResult, ProtocolError};
+use futures::StreamExt as _;
+use handle::P2PHandle;
+use libp2p::{
+    Multiaddr, PeerId, Swarm, SwarmBuilder, Transport,
+    core::{ConnectedPoint, muxing::StreamMuxerBox, transport::MemoryTransport},
+    gossipsub::{
+        Event as GossipsubEvent, Message, MessageAcceptance, MessageId, PublishError, Sha256Topic,
+    },
+    identity::ed25519::Keypair,
+    noise,
+    request_response::{self, Event as RequestResponseEvent},
+    swarm::SwarmEvent,
+    yamux,
+};
+use tokio::{
+    select,
+    sync::{broadcast, mpsc},
+    time::timeout,
+};
+use tokio_util::sync::CancellationToken;
+use tracing::{debug, error, info, instrument, trace, warn};
+
+use crate::{
+    commands::{Command, QueryP2PStateCommand},
+    events::Event,
+    operator_pubkey::P2POperatorPubKey,
+};
+
+mod behavior;
+mod codec_raw;
+pub mod errors;
+pub mod handle;
+
+/// Global topic name for gossipsub messages.
+// TODO(Velnbur): make this configurable later
+static TOPIC: LazyLock<Sha256Topic> = LazyLock::new(|| Sha256Topic::new("bitvm2"));
+
+/// Global MAX_TRANSMIT_SIZE for gossipsub messages.
+const MAX_TRANSMIT_SIZE: usize = 512 * 1024;
+
+/// Global name of the protocol
+// TODO(Velnbur): make this configurable later
+const PROTOCOL_NAME: &str = "/strata-bitvm2";
```

Comment:
All of this needs to be exposed to users and configurable.
It's hardcoded now.
We shouldn't tackle this in this PR, just pointing out.